nixui/options/ast.py

```diff
@@ -34,80 +35,30 @@ def get_full_node_string(node):
 #############
 def get_ast(file_path):
     ast_str = get_ast_str(file_path)
-    ast_str = ast_str.decode().replace('\\\"', 'EQ')  # TODO 3929: fix hack - can't parse \" properly
     return parse_ast_str(ast_str)
 
-
 def get_ast_str(file_path):
     res = subprocess.run(
         [
-            "dump-ast",
+            "nix_dump_cst_json",
             file_path
         ],
         stdout=subprocess.PIPE,
     )
     return res.stdout
 
-
 def parse_ast_str(ast_str):
-    # parsing the dumped ast string
-    tree = AST_DUMP_GRAMMAR.parse(ast_str)
-    root_node = NixVisitor().visit(tree)
-    assert len(root_node) == 1
-    return root_node[0]
+    return parse_ast_dict_node_or_token(json.loads(ast_str))
 
+def parse_ast_dict_node_or_token(d):
```

Comment:
Nit: this should be defined after `Node`, `Token` and `NumRange` are

Nit: We should call it `cst` not `ast` if the rust program returns a concrete syntax tree.

Reply:
or instead of cst, everywhere we could just say `syntax_tree`.

Reply:
https://github.com/nix-gui/nix-gui/pull/42/commits/01384311f0f41db6260f552da567370fff2f7acc